### PR TITLE
Set flag to install and start openvswitch (bsc#950798)

### PR DESCRIPTION
**Pull request by rhafer**
All compute nodes, even if we don't ourself create bridges on them need ovs
running when the ovs mech driver is selected.

### CODE REVIEW

**Comment by rhafer** [comment]
This needs to be merged together with https://github.com/crowbar/crowbar-core/pull/105
For this reason the ci will not be able to succeed testing this PR.

File: crowbar_framework/app/models/neutron_service.rb [hunk 1]
```diff
@@ -276,6 +276,14 @@ def update_ovs_bridge_attributes(attributes, node)
     if attributes["networking_plugin"] == "ml2"
       ml2_type_drivers = attributes["ml2_type_drivers"]
       if attributes["ml2_mechanism_drivers"].include?("openvswitch")
+        # This node needs the ovs packages installed and the service started
+        unless node.crowbar["network"]
+          node.crowbar["network"] = {}
+        end
```
Comment (vuntz) [inline crowbar_framework/app/models/neutron_service.rb]: `node.crowbar["network"] ||= {}` instead of the 3 lines?

**Comment by vuntz** [comment]
Do we need to do that for compute nodes also?

**Comment by rhafer** [comment]
> Do we need to do that for compute nodes also?

It's already there. NovaService actually calls into  NeutronService for that already. See: https://github.com/crowbar/crowbar-openstack/blob/master/crowbar_framework/app/models/nova_service.rb#L218

File: crowbar_framework/app/models/neutron_service.rb (hunk 1 above)
Comment (rhafer) [inline crowbar_framework/app/models/neutron_service.rb]: Good, question :). I'll change that.

**Comment by vuntz** [comment]
+1

**Comment by rhafer** [comment]
> This needs to be merged together with crowbar/crowbar-core#105

BTW, I was wrong with that. This one mergable without crowbar/crowbar-core#105. In fact it's better to merge this one before. That way the CI should be able to succeed for the other PR.

**Comment by rhafer** [comment]
I needed to rework the condition to disable "needs_openvswitch" a bit. It might have wrongly disabled the switch on compute nodes in some cases.

**Comment by jsuchome** [comment]
@rhafer so it's in do-not-merge state now?

**Comment by rhafer** [comment]
@jsuchome No. It's fixed already. Just requires review.

**Comment by jsuchome** [comment]
+1 (for the code; not tested)

File: crowbar_framework/app/models/neutron_service.rb [hunk 2]
```diff
@@ -308,6 +316,12 @@ def update_ovs_bridge_attributes(attributes, node)
         end
       end
     end
+    if ovs_bridge_networks.empty? && !ml2_mechanism_drivers.include?("openvswitch")
```
Comment (rhafer) [inline crowbar_framework/app/models/neutron_service.rb]: @vuntz still ok with this? This is the part I updated. I didn't have that ```&& !ml2_mechanism_drivers.include?("openvswitch")``` before, which would have caused the ```needs_openvswitch``` flag to be unset again on compute nodes :/.

Comment (vuntz) [inline crowbar_framework/app/models/neutron_service.rb]: Yes, +1

**Comment by aspiers** [comment]
Would be nice to have the full bug URL in the commit body ;-)

**Pull request merged**